### PR TITLE
Build: Add eslint-plugin-storybook to the repo

### PR DESCRIPTION
**Pull request by yannbf**
Issue: N/A

## What I did

This applies the [eslint-plugin-storybook](https://github.com/storybookjs/eslint-plugin-storybook) to the monorepo, as well as fixes for all of its recommended rules.

There are quite a few changes. I added some `@TODO` for discussions.

## How to test

- [x] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: examples/angular-cli/src/stories/core/parameters/bootstrap-options.stories.ts [hunk 1]
```diff
@@ -20,5 +20,3 @@ export default {
 } as Meta;
 
 export const WithPreserveWhitespaces: Story = (_args) => ({});
-
-WithPreserveWhitespaces.storyName = 'With Preserve Whitespaces';
```
Comment (kylegach) [inline examples/angular-cli/src/stories/core/parameters/bootstrap-options.stories.ts:24]: Oh, nice! You even made it work to catch the formatted case, not just the literal. 💯  👏

File: examples/web-components-kitchen-sink/src/stories/misc/script/script.stories.ts [hunk 2]
```diff
@@ -1,17 +1,19 @@
+/* eslint-disable storybook/hierarchy-separator */
+// @TODO: discuss about having dots in a meta title (that is not related to a separator)
```
Comment (tmeasday) [inline examples/web-components-kitchen-sink/src/stories/misc/script/script.stories.ts:2]: There are plenty of reasons why people might want to do this (e.g. "java-style" component names). I think people might just disable the rule entirely if they want to do something like that though.

File: examples/angular-cli/src/stories/core/parameters/bootstrap-options.stories.ts (hunk 1 above)
Comment (yannbf) [reply examples/angular-cli/src/stories/core/parameters/bootstrap-options.stories.ts:24]: Thank you!

**Review by shilman — APPROVED** [review]
Great job @yannbf 💯

File: examples/official-storybook/stories/other-dirname.stories.js [hunk 3]
```diff
@@ -1,10 +1,11 @@
+/* eslint-disable storybook/default-exports */
+// @TODO: can we delete this file? doesn't seem to make sense.
```
Comment (yannbf) [inline examples/official-storybook/stories/other-dirname.stories.js:2]: YES

File: examples/web-components-kitchen-sink/src/stories/misc/script/script.stories.ts (hunk 2 above)
Comment (yannbf) [reply examples/web-components-kitchen-sink/src/stories/misc/script/script.stories.ts]: Makes sense. Fixed by ignoring the dot in the eslint rule!

**Pull request merged**